mlir/lib/Dialect/Linalg/Utils/Utils.cpp

```diff
@@ -240,6 +240,475 @@ bool isReductionIterator(utils::IteratorType iteratorType) {
   return iteratorType == utils::IteratorType::reduction;
 }
 
+//===----------------------------------------------------------------------===//
+// Convolution matcher utilities
+//===----------------------------------------------------------------------===//
+
+/// Utility to match block body for linalg.pool* ops.
+template <typename... OpTypes>
+static bool bodyMatcherForPoolOps(Value yieldVal, Block *body) {
```

Comment:
How come we only check the body for pooling convs and neither for depthwise nor regular convs?